renku/cli/dataset.py

```diff
@@ -362,6 +374,108 @@ def remove(client, names):
     click.secho('OK', fg='green')
 
 
+@dataset.command('import')
+@click.argument('uri')
+@click.option('-n', '--name', help='Dataset name.')
+@click.option(
+    '-x',
+    '--extract',
+    is_flag=True,
+    help='Extract files before importing to dataset.'
+)
+@click.pass_context
+def import_(ctx, uri, name, extract):
+    """Import data from 3rd party provider."""
+    is_doi = DOIProvider.is_doi(uri)
+
+    if is_doi is False:
+        url = urlparse(uri)
+        if bool(url.scheme and url.netloc and url.params == '') is False:
+            raise BadParameter(
+                'Could not process {0}.\nValid formats are: DOI, URL'.
+                format(uri)
+            )
+
+    provider = None
+    if 'zenodo' in uri:
+        provider = ZenodoProvider()
+
+    try:
+        record = provider.find_record(uri, is_doi=is_doi)
+    except LookupError:
+        raise BadParameter(
+            'Could not process {0}.\nURI not found.'.format(uri)
+        )
+
+    files_ = record.get_files()
+    click.echo(
+        tabulate(
+            files_,
+            headers=OrderedDict((
+                ('checksum', None),
+                ('key', 'filename'),
+                ('size', None),
+                ('type', None),
+            ))
+        )
+    )
+
+    text_prompt = 'Do you wish to download this version?'
+    if record.last_version is False:
+        text_prompt = WARNING + 'Newer version found.\n' + text_prompt
+
+    if files_ and click.confirm(text_prompt):
+        data_folder = tempfile.mkdtemp()
+
+        pool = mp.Pool(mp.cpu_count())
```

Comment:
IMHO it should be configurable and probably not really depend on CPUs since it is network bounded - more CPUs != more network.

Reply:
yes, it definitely should not consume all of the cores. I would say make it half of the available cores by default and max of 4.

Reply:
Worker pool size does not directly determine how much CPU cores we consume. Those 2 things are not related in such a way. Look at Little's Law why this math works here. I will remove cpu_count from here since it confused both of you and will leave Python to do it's internal magic here since it will also rely on the same law.

Reply:
Sure I understand that -- but depending on what you are doing exactly and if the bandwidth allows it, you could consume 1 core per process. All I'm saying is that on a shared machine (such as a head node of an hpc cluster) it's typically not looked at favorably if one user spawns tons of processes so some modulation is needed here.

Reply:
```
pool_size = min(
            int(os.getenv('RENKU_POOL_SIZE', mp.cpu_count() // 2)), 4
)
pool = mp.Pool(pool_size)
```